### PR TITLE
Add attributes section to sign up request and optional attributes

### LINKED ISSUE #15: Allow to send "id" in sign up request
We need support a client-generated ID for for Hoodie. JSON API [explicitly supports client-generated resource IDs](http://jsonapi.org/format/#crud-creating-client-ids), so let’s put it into the spec, that it’s allowed, but not required of course

### PR DESCRIPTION
**Pull request by barretodaniel**
closes #15

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -250,14 +250,25 @@ or set account properties or destroy their accounts.
 
             Accept: application/vnd.api+json
 
+    + Attributes (object)
+        + data (object, required)
+            + type (string, required)
+            + attributes (object, required)
+                + username (string, required)
+                + password (string, required)
+                + createdAt (string, optional) - timestamp the user account was created on the client
```
Comment (barretodaniel) [inline apiary.apib]: To show the difference between optional and required request body attributes, it was necessary to add this. What other optional attributes are can be sent? Also, id inside `attributes` or outside like it is now?

Comment (gr2m) [reply apiary.apib]: I think I would leave out the `createdAt` attribute here, and set username & password to optional. They are just examples, the app implementing the spec can decide what attributes they want to use for their accounts.

**Review by gr2m — APPROVED** [review]
Looks good :) Thanks!

**Comment by gr2m** [comment]
Hey @barretodaniel, are you on twitter? We like giving shout outs to new contributors :) Only if you like of course

**Comment by barretodaniel** [comment]
I have a twitter but it's been a few months since I've used it lol. I
appreciate the gesture however. Feel free to mention my github if you like.

**Comment by barretodaniel** [comment]
BTW, we still have that other open issue. Are we okay to start implementing the server route?

**Comment by gr2m** [comment]
yes we are :) Thanks for your help Daniel :)

**Comment by gr2m** [comment]
https://twitter.com/hoodiehq/status/782671598255939584 :)

**Comment by barretodaniel** [comment]
Awwwwww :tada:

**Comment by barretodaniel** [comment]
Honestly, working on this has increased my comfort level with contributing tremendously. It's a nice feeling and I'm excited to continue contributing.

**Comment by gr2m** [comment]
that is really nice to hear, thanks Daniel! It’s really exactly the reason why we invest so much into being the most welcoming community possible :)

**Pull request merged**